ui/src/app/archived-workflows/components/archived-workflow-details/archived-workflow-details.tsx

```diff
@@ -247,6 +247,10 @@ export class ArchivedWorkflowDetails extends BasePage<RouteComponentProps<any>,
     }
 
     private openLink(link: Link) {
-        document.location.href = link.url.replace('${metadata.namespace}', this.state.workflow.metadata.namespace).replace('${metadata.name}', this.state.workflow.metadata.name);
```

Comment:
Looks good! Anyway we can refactor this into a new React component, e.g.` <LinkButton workflow={this.state.workflow}/>` to reduce duplication.

Reply:
or maybe better still:

```
<Links links={links} scope='workflow' object={this.status.workflow}/>
```

and

```
<Links links={links} scope='pod' object={node}/>
```

Reply:
Hmmm... I'm not familiar with React. Could you support me to refactor that?

Reply:
@alexec Buttons for `links` is rendered via `actionMenu` property of `Page` component which is from [argo-ui](https://github.com/argoproj/argo-ui). And `Page` component actually render buttons for `links` via [`Topbar` component](https://github.com/argoproj/argo-ui/blob/master/src/components/top-bar/top-bar.tsx#L98).

How can I refactor this code without modifying `react-ui`? It doesn't look easy. 😭

Reply:
ah - forget about it then